### PR TITLE
add gamepad support

### LINKED ISSUE #791: Controller support in launcher
### Role

I play Minecraft on the steamdeck and want to be able to start the game with the controller 

### Suggestion

add controller support on the launcher

### Benefit

I will be able to start a game with a controller on the Steamdeck game and desktop modes 

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

When you open the launcher on the Steamdeck the only way to select a game is on the touchscreen, but in desktop mode the screen is off, therefore it would be great if you could navigate the launcher's menu with a controller, or something with xinput.

### PR DESCRIPTION
**Pull request by Trial97**
fixes #791
no idea how to fix the CI but the code works for local :)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -532,6 +535,12 @@ else()
     message(STATUS "Using system ghc_filesystem")
 endif()
 add_subdirectory(libraries/qdcss) # css parser
+if(NOT SDL2_FOUND)
+    message(STATUS "Using bundled SDL2")
+    add_subdirectory(libraries/SLD) # toml parser
```
Comment (DioEgizio) [inline CMakeLists.txt]:
```suggestion
    add_subdirectory(libraries/SDL) # SDL (used for gamepad support)
```

**Comment by Trial97** [comment]
Status update:
- nix fails because it doesn't pull the submodules
- and all other stuff fails because it can't find the SDL2 header files

**Comment by Lutitious** [comment]
testing on windows with an xbox controller, the mouse is very jittery and behaves as a dpad even though it is being controlled by the analogue stick. Any launched window or menu causes the mouse to become unresponsive and the dpad selection also no longer works.

**Comment by Trial97** [comment]
Thinking about closing this in favor of other external tools that add this.

**Comment by ttang4299** [comment]
@Trial97 That's fair. Would it be built into the launcher? Just wondering if it would be streamlined for ease of use with Steam Deck users.

**Comment by Trial97** [comment]
@ttang4299 see https://github.com/PrismLauncher/PrismLauncher/issues/791#issuecomment-2143720678
But no I don't think there will be an integration with the launcher.

**Pull request closed without merging**